sdks/python/apache_beam/transforms/cy_combiners.py

```diff
@@ -308,3 +308,94 @@ class AnyCombineFn(AccumulatorCombineFn):
 
 class AllCombineFn(AccumulatorCombineFn):
   _accumulator_type = AllAccumulator
+
+MAX_LONG_10_FOR_LEADING_ZEROS = [19, 18, 18, 18, 18, 17, 17, 17, 16, 16, 16, 15,
+                                 15, 15, 15, 14, 14, 14, 13, 13, 13, 12, 12, 12,
+                                 12, 11, 11, 11, 10, 10, 10, 9, 9, 9, 9, 8, 8,
+                                 8, 7, 7, 7, 6, 6, 6, 6, 5, 5, 5, 4, 4, 4, 3, 3,
+                                 3, 3, 2, 2, 2, 1, 1, 1, 0, 0, 0]
+
+LONG_SIZE = 64
+
+def compare_to(x, y):
+  """return the sign bit of x-y"""
+  if x < y:
+    return 1
+  return 0
+
+def get_log10_round_to_floor(element):
+  number_of_leading_zeros = LONG_SIZE - element.bit_length()
+  y = MAX_LONG_10_FOR_LEADING_ZEROS[number_of_leading_zeros]
+  return y - compare_to(element, math.pow(10, y))
+
+class DistributionAccumulator(object):
+  """Distribution Counter:
+  contains value distribution statistics and methods for incrementing
+  """
+  def __init__(self):
+    self.min = 0
+    self.max = 0
+    self.count = 0
+    self.sum = 0
+    self.sum_of_squares = 0
+    """Histogram buckets of value counts for a distribution(1,2,5 bucketing)"""
+    self.buckets = []
+    """Starting index of the first stored bucket"""
+    self.first_bucket_offset = 0
+    """There are 3 buckets for every power of ten: 1, 2, 5"""
+    self.buckets_per_10 = 3
+
+  def add_input(self, element):
+    if element < 0:
+      return
+    self.min = min(self.min, element)
+    self.max = max(self.max, element)
+    self.count += 1
+    self.sum += element
+    self.sum_of_squares += math.pow(element, 2)
```

Comment:
element * element would be more efficient.

Reply:
Right, will fix in next revision. Thanks!